src/tools/compiletest/src/runtest.rs

```diff
@@ -3785,6 +3785,11 @@ impl<'test> TestCx<'test> {
         debug!(?support_lib_deps);
         debug!(?support_lib_deps_deps);
 
+        let mut host_dylib_env_paths = String::new();
+        host_dylib_env_paths.push_str(&cwd.join(&self.config.compile_lib_path).to_string_lossy());
+        host_dylib_env_paths.push(':');
+        host_dylib_env_paths.push_str(&env::var(dylib_env_var()).unwrap());
```

Comment:
I did wonder about that, but decided to leave it alone for the moment. I'll update these now.

Perhaps the `is_windows()` bit in `run_make_support::run` is also related? That seemed odd to me, but I don't want to touch Windows if it's working... :sweat_smile:

Reply:
https://github.com/rust-lang/rust/blob/f13f37fd7bbbc34c663311739176fe114ef51085/src/tools/run-make-support/src/run.rs#L15-L34

This bit is almost completely wrong, I did not know better at the time when this was written (I realize this is terribly broken *now*) 😆

Reply:
I'll put up a fix for this lol

Reply:
Ah, but a caveat: the `run` might need `TARGET_RPATH_DIR` in addition to `HOST_RPATH_DIR`, so it might be `set_host_rpath` but also setting target rpath.

Reply:
The original `RUN` only used the target path:
https://github.com/rust-lang/rust/blob/616a8f85f153033d798b583c0ad0431924a738b0/tests/run-make/tools.mk#L109-L110

Reply:
Oh I remember why: you're right, but conditionally right...

`RUN` accepts different environments depending on platform:

https://github.com/rust-lang/rust/blob/616a8f85f153033d798b583c0ad0431924a738b0/tests/run-make/tools.mk#L85-L116

Reply:
e.g. on windows it's `RUN = PATH="$(PATH):$(TARGET_RPATH_DIR)" $(EXECUTE)`...

Reply:
Safe to say, there's a lot of weird legacy there...

Reply:
I think then it's fine to leave `run()` as is, I'll go back to fix `run()` in a follow up PR. I don't want to block this PR for different reasons.

Reply:
Filed https://github.com/rust-lang/rust/issues/123832 to make sure I double check `run_make_support::run`